### PR TITLE
OCLOMRS-878: Previous Concept Form Validation Error Message Persists into New Custom Concept Form

### PR DESCRIPTION
**Pull request by jwnasambu**
# JIRA TICKET NAME:
https://issues.openmrs.org/browse/OCLOMRS-878

# Summary:
I added a return function to the effect. This function is to be called when the effect needs to be cleaned up.

### CODE REVIEW

File: src/apps/concepts/pages/CreateOrEditConceptPage.tsx [hunk 1]
```diff
@@ -120,6 +123,10 @@ const CreateOrEditConceptPage: React.FC<Props> = ({
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, []);
 
+  useEffect(() => {
+    return () => { resetConceptForm(); }
+}, []);
```
Comment (ibacher) [inline src/apps/concepts/pages/CreateOrEditConceptPage.tsx:128]:
```suggestion
  useEffect(() => resetConceptForm(),
   // eslint-disable-next-line react-hooks/exhaustive-deps
   []);
```

This is a slightly simpler definition, and we should disable the linter for this line as we are intentionally not putting anything in the dependency array.

File: src/apps/concepts/pages/CreateOrEditConceptPage.tsx [hunk 2]
```diff
@@ -236,7 +243,8 @@ const mapStateToProps = (state: AppState) => ({
 const mapActionsToProps = {
   retrieveDictionary: makeRetrieveDictionaryAction(true),
   retrieveConcept: retrieveConceptAction,
-  upsertConcept: upsertConceptAndMappingsAction
+  upsertConcept: upsertConceptAndMappingsAction,
+  resetConceptForm: resetConceptFormAction
```
Comment (ibacher) [inline src/apps/concepts/pages/CreateOrEditConceptPage.tsx:247]: Move the `resetConceptForm` line to the top of this block so it's in alphabetical order again.

**Review by ibacher — COMMENTED** [review]
Two small things, but this looks fine to me. Let's get some input from someone involved with OCL for OpenMRS, however.

**Comment by jwnasambu** [comment]
@ibacher  I have fixed the proposed changes. Thanks for review.

File: src/apps/concepts/pages/CreateOrEditConceptPage.tsx [hunk 3]
```diff
@@ -234,9 +241,10 @@ const mapStateToProps = (state: AppState) => ({
 });
 
 const mapActionsToProps = {
+  resetConceptForm: resetConceptFormAction,
   retrieveDictionary: makeRetrieveDictionaryAction(true),
   retrieveConcept: retrieveConceptAction,
-  upsertConcept: upsertConceptAndMappingsAction
+  upsertConcept: upsertConceptAndMappingsAction
```
Comment (ibacher) [inline src/apps/concepts/pages/CreateOrEditConceptPage.tsx:247]:
```suggestion
  upsertConcept: upsertConceptAndMappingsAction
```

**Review by ibacher — APPROVED** [review]

**Pull request merged**